ArxTypeTraits/type_traits.h

```diff
@@ -378,18 +388,17 @@ namespace std {
     using result_of = details::result_of<Sig>;
 
 
-    template <class T>
-    void swap(T& a, T& b)
-    {
-        T t = move(a);
-        a = move(b);
-        b = move(t);
-    }
+} } // namespace arx::std
 
-
-#ifndef ARX_TYPE_TRAITS_INITIALIZER_LIST_DEFINED
+// Initializer_list *must* be defined in std, so take extra care to only
+// define it when <initializer_list> is really not available (e.g.
+// ArduinoSTL is C++98 but *does* define <initializer_list>) and not
+// already defined (e.g. by ArxContainer).
+#if __has_include(<initializer_list>)
+#include <initializer_list>
+#elif !defined(ARX_TYPE_TRAITS_INITIALIZER_LIST_DEFINED)
 #define ARX_TYPE_TRAITS_INITIALIZER_LIST_DEFINED
-
+namespace arx_std {
```

Comment:
This should be `std` instead of `arx_std`?

Reply:
Yup, too eager search-replace :-) Pushed a fix just now.